### ISSUE TITLE
getConcreteMemoryAreaValue can't read out memory correctly in pin mode ?

### ISSUE THREAD

**Issue by KnoooW** [JonathanSalwan/Triton#632]
I cannot get correct values with getConcreteMemoryAreaValue in symproc callback:
1.c
====
```
void f(char *p, int *lenP) {
	printf("p:%s lenP:%x\n", p, *lenP) ;
}
int main() {
	char *p = "xxxxaazzz" ;
	int len = 0xaabbccdd ;
	f(p, &len) ;
}
```
=====

test.py
======
```

from triton  import *
from pintool import *
import sys

init = False

def cbeforeSymProc(instruction):
	global init

	if init == False:
		init = True
		buf = getConcreteRegisterValue(REG.RDI)
		psize = getConcreteRegisterValue(REG.RSI)
		print 'rdi: 0x%x rsi:0x%x'%(buf, psize)

		bytes_str = getConcreteMemoryAreaValue(buf, 4)
		print 'buf:', repr(bytes_str)

		bytes_str = getConcreteMemoryAreaValue(psize, 4)
		print 'psize:', repr(bytes_str)
	return


if __name__ == '__main__':

	# Set architecture
	setArchitecture(ARCH.X86_64)
	enableMode(MODE.ONLY_ON_TAINTED, True)
	#enableMode(MODE.ALIGNED_MEMORY, True)

	setAstRepresentationMode(AST_REPRESENTATION.PYTHON)
	startAnalysisFromSymbol('f')

	insertCall(cbeforeSymProc, INSERT_POINT.BEFORE_SYMPROC)

	# Run the instrumentation - Never returns
	runProgram()

```

with commands:
```
gcc 1.c -O0
triton ./test.py ./a.out
```

I get these:
```
rdi: 0x40062e rsi:0x7ffcc3d55cd4
buf: '\x00\x00\x00\x00'
psize: '\x00\x00\x00\x00'
p:xxxxaazzz lenP:aabbccdd

```

Why ?
buf should be read out as 'xxxx' and psize should be read as '\xdd\xcc\xbb\xaa'
but  I get all ZEROs with **getConcreteMemoryAreaValue**, it seems it doesn't work
properly ?

**Comment by JonathanSalwan** [comment]
Because their were not synchronized with Triton (never been read before). You have to use `pintool.getCurrentMemoryValue`.

**Comment by JonathanSalwan** [comment]
Actually we can consider this as a bug. I've patched this behavior in the `v0.6`.

**Comment by KnoooW** [comment]
thanks ! I think I just confused these APIs, may be add the API call in examples can do some help :)
